### ISSUE TITLE
Autobuild fails for .NET 9

### ISSUE THREAD

**Issue by XFox111** [github/codeql-action#2625]
Microsoft released .NET 9 STS on November 12, 2024, and it's now in active support phase.

I've added support for .NET 9 in my project, but once I tried to run CodeQL on it, it failed to autobuild because no .NET 9 SDK was installed.

```log
Run github/codeql-action/autobuild@v3
  with:
    token: ***
    matrix: {
    "language": "csharp"
  }
  env:
    <...>
Attempting to automatically build csharp code
  Picked up JAVA_TOOL_OPTIONS:  -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false
  /opt/hostedtoolcache/CodeQL/2.19.3/x64/codeql/csharp/tools/autobuild.sh 
  CodeQL C# autobuilder
  .NET SDK:
   Version:           8.0.404
   Commit:            7b190310f2
   Workload version:  8.0.400-manifests.996cfe54
   MSBuild version:   17.11.9+a69bbaaf5
  
  Runtime Environment:
   OS Name:     ubuntu
   OS Version:  22.04
   OS Platform: Linux
   RID:         linux-x64
   Base Path:   /usr/share/dotnet/sdk/8.0.404/
  
  .NET workloads installed:
  Configured to use loose manifests when installing new manifests.
  There are no installed workloads to display.
  
  Host:
    Version:      8.0.11
    Architecture: x64
    Commit:       9cb3b725e3
  
  .NET SDKs installed:
    6.0.428 [/usr/share/dotnet/sdk]
    7.0.410 [/usr/share/dotnet/sdk]
    8.0.404 [/usr/share/dotnet/sdk]
  
  .NET runtimes installed:
    Microsoft.AspNetCore.App 6.0.36 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
    Microsoft.AspNetCore.App 7.0.20 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
    Microsoft.AspNetCore.App 8.0.11 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
    Microsoft.NETCore.App 6.0.36 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
    Microsoft.NETCore.App 7.0.20 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
    Microsoft.NETCore.App 8.0.11 [/usr/share/dotnet/shared/Microsoft.NETCore.App]
  
  Other architectures found:
    None
  
  Environment variables:
    Not set
  
  global.json file:
    Not found

<...>
```

**Comment by redsun82** [comment]
👋 @XFox111 

It appears that .NET 9 is not yet available on ubuntu 22.04 and 24.04 images (sources [here](https://github.com/actions/runner-images/blob/ubuntu22/20241124.1/images/ubuntu/Ubuntu2204-Readme.md#net-tools) and [here](https://github.com/actions/runner-images/blob/ubuntu22/20241124.1/images/ubuntu/Ubuntu2404-Readme.md#net-tools)). As far as I could see that version only comes preinstalled on Windows 2022 runners.

You should probably try using [actions/setup-dotnet](https://github.com/marketplace/actions/setup-net-core-sdk) to setup the required .NET toolchain if it's not included in the image.

**Comment by XFox111** [comment]
Hi @redsun82,

Ah, I see. I thought that autobuild installs all necessary tools itself, instead of relying on runner environment. Sorry to bother :)
